### ISSUE TITLE
SwiftLint tool should fail builds if accessibility tags are not present for UIControls

### ISSUE THREAD

**Issue by KarthikaThevan** [realm/SwiftLint#3484]
Hi Team,

We are having ebook reading application developed in IOS platform. We want to use swiftLint tool to check whether accessibility enabled for all UIControls which are in storyboard as well as programmatically initialised controls.

We had a walk through for SwiftLint rules, but we couldn’t find the rule to check the accessibility enabled/ disabled controls.

**Our requirement:** We are supporting voice over for visually disabled people. Accessibility should be enabled for all UI controls (eg : UIButton, UILabel… etc) in application to support voice over.

**Use case** : When we make a build, if the controls doesn’t have accessibility enabled true property, then the build should fail in Xcode. SwiftLint tool should fail builds if accessibility tags are not present for UIControls in the application.

If SwiftLint is supporting this feature, could you explain how to use it. If not, Could you take it as feature request?

**Comment by KarthikaThevan** [comment]
@marcelofabri Could you help us to resolve this? If the feature is not available already, could you guide how to add new feature in SwiftLint.

**Comment by KarthikaThevan** [comment]
Hi Team,

We are having ebook reading application developed in IOS platform. We want to use swiftLint tool to check whether accessibility enabled for all UIControls which are in storyboard as well as programmatically initialised controls.

Our requirement: We are supporting voice over for visually disabled people. Accessibility should be enabled for all UI controls (eg : UIButton, UILabel… etc) in application to support voice over.

Use case : When we make a build, if the controls doesn’t have accessibility enabled true property, then the build should fail in Xcode. SwiftLint tool should fail builds if accessibility tags are not present for UIControls in the application.

Kindly guide us how to get all UIElements in the application and how to check the property values of the UIElement instance.
For example, get all the UILabels and check whether the UILabels are set as IsAccessibilityElement is True. Likewise, we need to get all UIElements and check the IsAccessibilityElement property value for each UIElement instance.

Could you please suggest on the above so that, we can try our best to implement it in SwiftLint.

**Comment by jpsim** [comment]
Looks like you also opened https://github.com/realm/SwiftLint/issues/3508 about the same thing, please see my comment there.
